src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -209,63 +261,43 @@ private static boolean longestCommonSubsequenceCheck(String one, String two) {
    *      Similarity</a>
    */
   private static boolean jaroWinklerSimilarityCheck(String one, String two) {
-    return JWS.apply(one, two) > METRIC_THRESHOLD;
+    return JWS.apply(one, two) > JWS_METRIC_THRESHOLD;
   }
 
   /**
-   * Checks if one of the cpeXXUri contains the project's name.
+   * Checks if one of the cpeXXUri contains the project's and organization's names. If the check is
+   * a success, it returns true immediately. If it could not match then check if the list of
+   * references contains any information.
    * 
    * @param cpeMatch of type {@link CpeMatch}.
+   * @param referenceMatch Indicates if there is a reference URL similar to project URL.
+   * @param project of type {@link GitHubProject}.
    * @return True if one of the cpeXXUri contains the project's name. Otherwise False.
    */
-  private boolean cpeUriCheck(CpeMatch cpeMatch) {
-    return match(productFrom23Uri(cpeMatch.getCpe23Uri()))
-        || match(productFrom22Uri(cpeMatch.getCpe22Uri()));
-  }
+  private static boolean projectCheck(CpeMatch cpeMatch, boolean referenceMatch,
+      GitHubProject project) {
+    CpeUri cpeUri = cpeMatch.getCpeUri();
 
-  /**
-   * Returns product name from {@link CpeMatch#getCpe23Uri()}.
-   */
-  private static String productFrom23Uri(String cpe23Uri) {
-    return stringAtIndex(cpe23Uri, ":", 4);
-  }
+    boolean productMatch = match(cpeUri.getProduct(), project.name());
+    boolean vendorMatch = match(cpeUri.getVendor(), project.organization().name());
 
-  /**
-   * Returns product name from {@link CpeMatch#getCpe22Uri()}.
-   */
-  private static String productFrom22Uri(String cpe22Uri) {
-    return stringAtIndex(cpe22Uri, ":", 3);
-  }
-
-  /**
-   * Returns the substring present at the index after the split by the given delimiter pattern.
-   * 
-   * @param content The String to be split.
-   * @param pattern The delimiter to be used.
-   * @param index The index of the sub-string which should be returned.
-   * @return The sub-string if the index is valid and the content is not null. Otherwise returns an
-   *         empty string.
-   */
-  private static String stringAtIndex(String content, String pattern, int index) {
-    if (content != null) {
-      String[] splitContent = content.split(pattern);
-      if (splitContent.length > index) {
-        return splitContent[index];
-      }
-    }
-
-    return StringUtils.EMPTY;
+    // Check if product's name matches the project's name. Otherwise at least one reference URL
+    // matches with the project's URL (This check is considered only when the product's or vendor's
+    // name check fails).
+    return productMatch || (referenceMatch && !(productMatch || vendorMatch));
```

Comment:
In the expression `(referenceMatch && !(productMatch || vendorMatch))`, `productMatch` is always `false`. Otherwise, the whole expression would be already `true`. It means that the expression becomes `(referenceMatch && !vendorMatch)`. It means that vendors have to be different. I am still not sure about this logic. It may work fine with the test data but this logic looks wrong in general.

If you'd like to keep it as-is, please simplify the expression and update the comment. We'll see how it works on other projects.